core/modules/widgets/transclude.js

```diff
@@ -97,13 +99,36 @@ TranscludeWidget.prototype.makeRecursionMarker = function() {
 	output.push("}");
 	return output.join("");
 };
+	
+TranscludeWidget.prototype.hasSourceTextChanged = function() {
+	var tiddler,
+			parserType,
```

Comment:
@pmario as mentioned in the OP this is just some quick code I hacked together on a touchscreen to help with discussion, it needs clean up. 

Where feedback would be helpful is in the approach. The `hasSourceTextChanged` method has a lot in common with [wiki.parseTextReference](https://github.com/Jermolene/TiddlyWiki5/blob/8fbf52e419e71d726ea32b6c44e3ccfc4245d825/core/modules/wiki.js#L939) which the `transclude` widget calls during rendering. It would have been good to use the same code in both places to ensure consistency between them, but I am not sure there is an elegant and backwards compatible way to refactor `parseTextReference` to achieve this. Thoughts?

Reply:
I don't think, it has enough in common to change the `wiki.parseTextReference` ... and there are no tests for `wiki.parseTextReference` atm. So you'd need to create tests for all code paths first.